Problem building tests

When I try to build the tests, I get the error messages below.

@manuelmessner  said something about having had a similar problem, maybe you can describe your fix in a little more detail?

Maybe this is sth. that needs to be fixed in general.

``` bash
cd /home/nadja/waysome/tests/check && /usr/bin/cmake -E cmake_link_script CMakeFiles/check_action.test.dir/link.txt --verbose=1
/usr/bin/cc   -std=gnu99 -pthread -g    CMakeFiles/check_action.test.dir/action/action_tests.c.o  -o check_action.test -rdynamic -Wl,-Bstatic -lcheck -Wl,-Bdynamic 
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check.o): In function `tcase_create':
(.text+0x194): undefined reference to `floor'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check.o): In function `tcase_create':
(.text+0x1a8): undefined reference to `floor'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check.o): In function `tcase_set_timeout':
(.text+0x435): undefined reference to `floor'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check.o): In function `tcase_set_timeout':
(.text+0x449): undefined reference to `floor'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check_run.o): In function `srunner_run':
(.text+0x8e5): undefined reference to `timer_create'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check_run.o): In function `srunner_run':
(.text+0x92a): undefined reference to `timer_settime'
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libcheck.a(check_run.o): In function `srunner_run':
(.text+0x959): undefined reference to `timer_delete'
collect2: error: ld returned 1 exit status
tests/check/CMakeFiles/check_action.test.dir/build.make:86: recipe for target 'tests/check/check_action.test' failed
make[3]: *** [tests/check/check_action.test] Error 1
make[3]: Leaving directory '/home/nadja/waysome'
CMakeFiles/Makefile2:718: recipe for target 'tests/check/CMakeFiles/check_action.test.dir/all' failed
make[2]: *** [tests/check/CMakeFiles/check_action.test.dir/all] Error 2
make[2]: Leaving directory '/home/nadja/waysome'
CMakeFiles/Makefile2:726: recipe for target 'tests/check/CMakeFiles/check_action.test.dir/rule' failed
make[1]: *** [tests/check/CMakeFiles/check_action.test.dir/rule] Error 2
make[1]: Leaving directory '/home/nadja/waysome'
Makefile:117: recipe for target 'tests/check/CMakeFiles/check_action.test.dir/rule' failed
make: *** [tests/check/CMakeFiles/check_action.test.dir/rule] Error 2
```

--- a/tests/check/CMakeLists.txt
+++ b/tests/check/CMakeLists.txt
@@ -6,7 +6,7 @@ include_directories(
     ${CHECK_INCLUDE_DIRS}
 )
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pthread")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pthread -lm -lrt")
 
 #
 # This variable will hold all the tests

